### PR TITLE
Add some C++ code along with BUILD generator

### PR DESCRIPTION
**Pull request by alexeagle**
Running `bazel configure` produces complete BUILD files, I've been testing it with

`find speller -name BUILD.bazel | xargs rm && bazel configure ; bazel test speller/...`

I'm putting up some public docs right now about this "starlark Gazelle extensibility feature" we built-in to Aspect CLI. Also have a talk about it coming up and rehearsing Friday morning.

### CODE REVIEW

**Comment by kylecordes** [comment]
I did not look to why these all failed in the CI. Probably something simple, I would guess.

Key bit about the BUILD files: Some of the BUILD files in the repo I was using have comments that explain the meaning of various parts. These have proven **very** useful in working with student groups. Fortunately, if the comments are brought over here they should be preserved by Gazelle.

File: .aspect/cli/rules_cc.star [hunk 1]
```diff
@@ -0,0 +1,95 @@
+aspect.register_rule_kind("cc_binary", {
```
Comment (kylecordes) [inline .aspect/cli/rules_cc.star:1]: I've not dug into Aspect CLI that deeply yet, so I don't know what these bits are for.

File: cli/BUILD.bazel [hunk 2]
```diff
@@ -1,5 +1,5 @@
-load("@rules_python//python:defs.bzl", "py_binary")
 load("@pip//:requirements.bzl", "requirement")
+load("@rules_python//python:defs.bzl", "py_binary")
```
Comment (kylecordes) [inline cli/BUILD.bazel:2]: I always have a bit of anguish around changes like this. It was semantically better before with the base language before the dependency management system, but it is syntactically more consistent now to have them sorted.

File: .aspect/cli/rules_cc.star (hunk 1 above)
Comment (kylecordes) [reply .aspect/cli/rules_cc.star:1]: Ah, I see the comment in Slack now... this is likely part of the new thing for your upcoming talk, so it makes sense that I haven't heard of this stuff yet.

Comment (alexeagle) [reply .aspect/cli/rules_cc.star:1]: yeah this is "what if you could write a Gazelle extension in a single interpreted script that lives in your repo".

You can see how it already handles all the logic needed here. We'll have a better story for sharing them at some point, but I'm betting the rules_cc maintainers will reject a PR adding it.

Comment (kylecordes) [reply .aspect/cli/rules_cc.star:1]: Could be challenging to get rule maintainers broadly to be interested in accepting this kind of thing. Maybe each could be packaged in a module, placed in to the BCR? Then all the version matching and discoverability is outsourced there.

Comment (alexeagle) [reply .aspect/cli/rules_cc.star:1]: That is possible. We don't even have the ability to `load` from external starlark files in our Gazelle-hosted starlark interpreter yet, so it's still an open design problem.

**Pull request merged**